enhancements/authentication/allow-users-to-manage-their-own-tokens.md

```diff
@@ -0,0 +1,153 @@
+---
+title: neat-enhancement-idea
+authors:
+  - "@stlaz"
+reviewers:
+  - "@sttts"
+  - "@deads2k"
+  - "@marun"
+  - "@vareti"
+approvers:
+  - "@sttts"
+  - "@deads2k"
+creation-date: 2020-10-22
+last-updated: 2020-10-22
+status: implementable
+see-also:
+- [enhancement: Secure OAuth Resource Storage](https://github.com/openshift/enhancements/pull/323)
+replaces:
+superseded-by:
+
+---
+
+# Allow Users To Manage Their Own Tokens
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+While users are quite capable of retrieving access tokens to be able to
+work with OpenShift services, their capabilities to see the tokens they
+created and to handle them is a bit lacking. This enhancement describes
+how the self-management of users' access tokens can be improved.
+
+## Motivation
+
+It is quite easy for users to retrieve tokens but it is also not so hard
+for them to be able to lose access to them - people don't usually search
+their cookies for the value of the token, or don't keep history of the
+tokens they retrieve by issuing the `oc login` command multiple times.
+
+### Goals
+
+Allow users to:
+- list their own tokens
+- view details of a specific token
+- delete their own tokens
+
+### Non-Goals
+
+- enable handling sessions of services that use the OpenShift access tokens
+
+## Proposal
+
+### User Stories [optional]
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+#### Story 1
+A user is able to see all of the tokens that were issued for them, including
+the name of the OAuth2 client that requested it and the scopes of it, making
+it easier for them to audit their accesses.
+
+#### Story 2
+A user gets a friendlier way to remove their own tokens,
+`oc delete oauthaccesstoken <sha256-tokenhash>` uses generic deletion code
+which attempts to `list` the token after its deletion, and that ends up
+with a `forbidden` message since the user is usually not allowed to `list`
+their tokens.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The OAuth API server gets a new API endpoint - `personaltokens`. This endpoint
```

Comment:
my vote goes to usersoauthaccesstokens

Reply:
It would have been ideal if `oc get oauthaccesstokens` listed just the access tokens of requesting user. But I guess that would not be possible without breaking the API.

+1 for useroauthaccesstokens

Reply:
doesn't quite roll off the tongue, but describes the resource more clearly

Reply:
> It would have been ideal if `oc get oauthaccesstokens` listed just the access tokens of requesting user

On the other hand, that might make it hard for the administrator to list all the tokens.